### PR TITLE
fix(inputs.upsd): add additional fields

### PR DESCRIPTION
**Pull request by powersj**
This adds some additional numeric fields around the battery runtime,
input transfer, and ups delay.

Additionally, this will grab the battery date (date it was changed)
and battery date mfr (date it was mrf) as two different fields.

fixes: #12517

### CODE REVIEW

**Comment by srebhan** [comment]
@powersj can you please rebase this PR to resolve the conflicts first!?

**Comment by powersj** [comment]
> @powersj can you please rebase this PR to resolve the conflicts first!?

done!

File: plugins/inputs/upsd/upsd.go [hunk 1]
```diff
@@ -73,24 +73,32 @@ func (u *Upsd) gatherUps(acc telegraf.Accumulator, name string, variables []nut.
...
+		"ups_status":       metrics["ups.status"],
+
 		//Compatibility with apcupsd metrics format
 		"time_left_ns": timeLeftS * 1_000_000_000,
-		"ups_status":   metrics["ups.status"],
 	}
 
 	floatValues := map[string]string{
 		"battery_charge_percent":  "battery.charge",
+		"battery_runtime":         "battery.runtime",
```
Comment (srebhan) [inline plugins/inputs/upsd/upsd.go:87]: Isn't this reported in `"time_left_ns"` already?

**Review by srebhan — COMMENTED** [review]
Looks good. Only one question regarding `battery_runtime` as this is already reported as `time_left_ns` IMO...

File: plugins/inputs/upsd/upsd.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/upsd/upsd.go]: The comment says it is for compatibility to the apcupsd plugin.

**Review by srebhan — APPROVED** [review]
LGTM. Thanks for adding those fields @powersj!

**Pull request merged**